File: CSharp-SQLite-APP/SQLiteFunctions/SQLiteFunctions/SQLiteFunctions/Functions/Dec2Bin.cs
using System;
using System.Data.SQLite;

namespace SQLiteFunctions.Functions
{
    [SQLiteFunction(Name = "dec2bin", Arguments = 1, FuncType = FunctionType.Scalar)]
    public class Dec2Bin : SQLiteFunction
    {
        public string ConstructCommand()
        {
            Console.Write("Decimal value: ");
            var input = Console.ReadLine();

            return $"dec2bin({input})";
        }

        private string DecToBin(int number)
        {
            string bin = "";
            string answer = "";

            if (number == 0) return "0";
            while (number > 0)
            {
                if (bin.Length == 0)
                {
                    if (number % 2 != 0) bin = bin.Insert(0, "1");
                    else bin = bin.Insert(0, "0");
                    number /= 2;
                    continue;
                }
                if (number % 2 != 0) bin = bin.Insert(bin.Length - 1, "1");
                else bin = bin.Insert(bin.Length - 1, "0");
                number /= 2;
            }
            for (int i = bin.Length - 1; i >= 0; i--)
            {
                if (answer.Length == 0)
                {
                    answer = answer.Insert(0, bin[i] + "");
                    continue;
                }
                answer = answer.Insert(answer.Length - 1, bin[i] + "");
            }

            return answer;
        }

        public override object Invoke(object[] args)
        {
            return DecToBin(int.Parse(args[0].ToString()));
        }
    }
}

File: Java-Oracle/dec2bin/Dec2Bin.java
import java.lang.System;

public class Dec2Bin {
    public static void main(String[] args) {
        System.out.println(DecToBin(0));
    }

    public static String DecToBin(int number) {
        String bin = "";
        String answer = "";
        if (number == 0) return "0";
        while(number > 0) {
            if(number % 2 != 0) bin = bin.concat("1");
            else bin = bin.concat("0");
            number /= 2;
        }
        for(int i = bin.length() - 1; i >= 0; i--) {
            answer = answer.concat(bin.charAt(i) + "");
        }

        return answer;
    }
}